lib/corsica.ex

```diff
@@ -277,7 +281,9 @@ defmodule Corsica do
   # Plug callbacks.
 
   def init(opts) do
-    sanitize_opts(opts)
+    opts
+    |> sanitize_opts
+    |> require_origins
```

Comment:
The check for `:origins` should be performed in `sanitized_opts/1`. That is, `sanitize_opts/1` should return valid options if the incoming keyword list is valid, otherwise raise. A `%Corsica.Options{}` struct should never be invalid.